### PR TITLE
xds: implement xDS server validations

### PR DESCRIPTION
**Pull request by sanjaypujare**

### CODE REVIEW

File: xds/src/test/java/io/grpc/xds/ServerXdsClientTest.java
Comment (voidzcy) [inline xds/src/test/java/io/grpc/xds/ServerXdsClientTest.java:98]: What's happening here? Is it just waiting for #7969 to be merged?

Comment (sanjaypujare) [reply xds/src/test/java/io/grpc/xds/ServerXdsClientTest.java:98]: Yes. After that merge this whole file is going to go away

**Review by dapengzhang0 — APPROVED** [review]
LGTM

**Comment by voidzcy** [comment]
Well, it's merged... I haven't got time to look at it...

**Pull request merged**
